File: CourseWorks/MHike/app/src/main/java/greenwich/comp1786/m_hike/model/Observation.java
package greenwich.comp1786.m_hike.model;

public class Observation {
    private int id;
    private int hikeId;
    private String observation;
    private String time;
    private String comment;

    public Observation(int id, int hikeId, String observation, String time, String comment) {
        this.id = id;
        this.hikeId = hikeId;
        this.observation = observation;
        this.time = time;
        this.comment = comment;
    }

    public Observation() {}

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getHikeId() {
        return hikeId;
    }

    public void setHikeId(int hikeId) {
        this.hikeId = hikeId;
    }

    public String getObservation() {
        return observation;
    }

    public void setObservation(String observation) {
        this.observation = observation;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }
}

File: CourseWork/mhike_android/lib/models/observation.dart
class Observation {
  int? id;
  int hikeId;
  String observation, time;
  String? comment;

  Observation({
    this.id,
    required this.hikeId,
    required this.observation,
    required this.time,
    this.comment,
  });

  Map<String, dynamic> toMap() => {
    'id': id,
    'hikeId': hikeId,
    'observation': observation,
    'time': time,
    'comment': comment,
  };

  static Observation fromMap(Map<String, dynamic> m) => Observation(
    id: m['id'],
    hikeId: m['hikeId'],
    observation: m['observation'],
    time: m['time'],
    comment: m['comment'],
  );
}
